### PR TITLE
Loops / it-academy.by / M-JC1-66-20 / Oxana Peregud / Done Tasks 01-10

### PR DESCRIPTION
**Pull request by OxanaPeregud**
## Pull request state

Please indicate actual status for PR:
- [x] on review
- [ ] reviewed
- [ ] fixed
- [ ] on approve
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task01.java [hunk 1]
```diff
@@ -31,6 +31,11 @@ public static void main(String[] args) {
     static String generateNumbersColumn(int numberRows) {
         //TODO
         // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+
+        StringBuilder numbersColumn = new StringBuilder();
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/javabasics/practice/loops/Task01.java:35]: Why are you using `StringBuilder`?)

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task02.java [hunk 2]
```diff
@@ -34,6 +34,16 @@ public static void main(String[] args) {
     static String generateNumbersColumn(int numberRows, boolean isEnableHeaderRow) {
         //TODO
         // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+
+        StringBuilder numbersColumn;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/javabasics/practice/loops/Task02.java:38]: Why are you using `StringBuilder`?)

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task06.java [hunk 3]
```diff
@@ -36,6 +36,19 @@ public static void main(String[] args) {
     static String convertToAccountingFormat(long amount) {
         //TODO
         // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+
+        StringBuilder accountingFormat = new StringBuilder();
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/javabasics/practice/loops/Task06.java:40]: Why are you using `StringBuilder`?)

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task08.java [hunk 4]
```diff
@@ -40,11 +40,27 @@ public static void main(String[] args) {
...
+     * @return price, где формат вывода одной записи '${numberItems} - ${cost}'. Для ${cost} использовать точность до
+     * 2 знаков после вещественного разделителя:
+     * <code>BigDecimal.valueOf(currentPriceAllItemsWithSell).setScale(2, RoundingMode.HALF_UP).doubleValue()</code>
      */
     static String generateTotalPriceList(int startNumberItems, double startPriceAllItems, int differentialNumberItems, int sizeTotalPrice) {
         //TODO
         // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+
+        StringBuilder totalPriceList = new StringBuilder();
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/javabasics/practice/loops/Task08.java:51]: Why are you using `StringBuilder`?)

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task08.java [hunk 5]
```diff
@@ -40,11 +40,27 @@ public static void main(String[] args) {
...
+        for (int i = 1; i <= sizeTotalPrice; i++) {
+            if (i == 1) {
+                totalPriceList.toString();
+            } else
+                startNumberItems = startNumberItems + differentialNumberItems;
+            cost = startNumberItems * itemPrise - (startNumberItems * itemPrise * totalSell / 100);
+            cost = Math.round(100.0 * cost) / 100.0;
+            totalPriceList.append(startNumberItems).append(" - ").append(cost).append("\n");
+        }
+        return totalPriceList.toString().replaceAll("[\n\r]$", "");
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/javabasics/practice/loops/Task08.java:64]: Why are you using Regex?)

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task09.java [hunk 6]
```diff
@@ -44,11 +44,31 @@ public static void main(String[] args) {
...
+     * '${numberItems} - ${cost} with sell ${totalSell}%'. Для ${cost} использовать точность до 2 знаков после
+     * вещественного разделителя:
+     * <code>BigDecimal.valueOf(currentPriceAllItemsWithSell).setScale(2, RoundingMode.HALF_UP).doubleValue()</code>
      */
     static String generateTotalPriceList(int startNumberItems, double startPriceAllItems, int differentialNumberItems, double differentialSell, int sizeTotalPrice) {
         //TODO
         // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+
+        StringBuilder totalPriceList = new StringBuilder();
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/javabasics/practice/loops/Task09.java:55]: Why are you using `StringBuilder`?)

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task09.java [hunk 7]
```diff
@@ -44,11 +44,31 @@ public static void main(String[] args) {
...
+            if (i == 1) {
+                totalPriceList.toString();
+            } else
+            startNumberItems = startNumberItems + differentialNumberItems;
+            totalSell = differentialSell * (i - 1);
+            cost = startNumberItems * itemPrise - (startNumberItems * itemPrise * totalSell / 100);
+            cost = Math.round(100.0 * cost) / 100.0;
+            totalPriceList.append(startNumberItems).append(" - ").append(cost).append(" with sell ").append(totalSell).append("%").append("\n");
+        }
+        return totalPriceList.toString().replaceAll("[\n\r]$", "");
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/javabasics/practice/loops/Task09.java:69]: Why are you using Regex?)

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task10.java [hunk 8]
```diff
@@ -13,7 +13,7 @@
  *
  * @author Dmitry Rakovets
  */
-class Task10 {
+class Task10<i> {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/javabasics/practice/loops/Task10.java:16]: What is it?

**Review by rakovets — COMMENTED** [review]
Good job but why are you using `StringBuilder`?)

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task01.java (hunk 1 above)
Comment (OxanaPeregud) [reply src/main/java/com/rakovets/course/javabasics/practice/loops/Task01.java:35]: To answer your question - the String can't be changed once it is created, but StringBuilder can be expanded. It does not create a new object in the memory and that's why doesn’t have the extra garbage left over. So StringBuilder is faster.

As to my knowlegde, the StringBuilder class is recommended for use with an unknown number of operations and changes on lines during program execution and also when the application is expected to do many such operations. Therefore, it seems to me that StringBuilder is more suitable for loops.

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task02.java (hunk 2 above)
Comment (OxanaPeregud) [reply src/main/java/com/rakovets/course/javabasics/practice/loops/Task02.java:38]: See above

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task06.java (hunk 3 above)
Comment (OxanaPeregud) [reply src/main/java/com/rakovets/course/javabasics/practice/loops/Task06.java:40]: See above

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task08.java (hunk 4 above)
Comment (OxanaPeregud) [reply src/main/java/com/rakovets/course/javabasics/practice/loops/Task08.java:51]: See above

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task08.java (hunk 5 above)
Comment (OxanaPeregud) [reply src/main/java/com/rakovets/course/javabasics/practice/loops/Task08.java:64]: I couldn't think of anything else to remove the space at the end of the string. If you know a better way, please, do tell)

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task09.java (hunk 6 above)
Comment (OxanaPeregud) [reply src/main/java/com/rakovets/course/javabasics/practice/loops/Task09.java:55]: See above

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task09.java (hunk 7 above)
Comment (OxanaPeregud) [reply src/main/java/com/rakovets/course/javabasics/practice/loops/Task09.java:69]: See above

File: src/main/java/com/rakovets/course/javabasics/practice/loops/Task10.java (hunk 8 above)
Comment (OxanaPeregud) [reply src/main/java/com/rakovets/course/javabasics/practice/loops/Task10.java]: My mistake - removed

**Review by rakovets — COMMENTED** [review]
Approved

**Pull request closed without merging**